src/vorta/views/extract_dialog.py

```diff
@@ -254,11 +176,103 @@ def parse_json_lines(lines, model: "ExtractTree"):
 # ---- Sorting ---------------------------------------------------------------
 
 
-class ExtractSortProxyModel(FileTreeSortProxyModel):
+class ExtractSortProxyModel(FileTreeSortFilterProxyModel):
     """
     Sort a ExtractTree model.
     """
 
+    def __init__(self, parent=None) -> None:
+        super().__init__(parent)
+
+    @staticmethod
+    def parse_date(value: str) -> Tuple[str, datetime]:
+        """
+        Parse the date string into a tuple of two values.
+        """
+
+        comparison_sign = ['<', '>', '<=', '>=', '=']
+
+        if not any(value.startswith(unit) for unit in comparison_sign):
+            raise argparse.ArgumentTypeError("Invalid date format. Supported comparison signs: <, >, <=, >=, =")
+
+        if value[1] == '=':
+            date = value[2:]
+            sign = value[:2]
+        else:
+            date = value[1:]
+            sign = value[:1]
+
+        try:
+            date = datetime.strptime(date, '%Y-%m-%d')
+        except ValueError:
+            raise argparse.ArgumentTypeError("Invalid date format. Must be YYYY-MM-DD.")
+
+        return (sign, date)
+
+    @classmethod
+    def valid_date_range(cls, value: str) -> List[Tuple[str, datetime]]:
+        """Parse the date range string."""
+        date = value.split(',')
+
+        if len(date) == 1:
+            return [cls.parse_date(date[0])]
+        elif len(date) == 2:
+            return [
+                cls.parse_date(date[0]),
+                cls.parse_date(date[1]),
+            ]
+        else:
+            raise argparse.ArgumentTypeError("Invalid date format. Can only accept two values.")
+
+    def get_parser(self):
+        """Add Extract view specific arguments to the parser."""
+        parser = super().get_parser()
+
+        health_group = parser.add_mutually_exclusive_group()
+        health_group.add_argument(
+            "--healthy", default=None, action="store_true", dest="healthy", help="Match only healthy items."
+        )
+        health_group.add_argument(
+            "--unhealthy", default=None, action="store_false", dest="healthy", help="Match only unhealthy items."
+        )
+
+        parser.add_argument(
+            "--last-modified",
+            type=self.valid_date_range,
+            help="Match by last modified date.",
+        )
+
+        return parser
+
+    def filterAcceptsRow(self, sourceRow: int, sourceParent: QModelIndex) -> bool:
+        """
+        Return whether the row should be accepted.
+        """
+
+        if not self.searchPattern:
+            return True
+
+        if not super().filterAcceptsRow(sourceRow, sourceParent):
+            return False
+
+        model = self.sourceModel()
+        item = model.index(sourceRow, 0, sourceParent).internalPointer()
+
+        if self.searchPattern.healthy is True and not item.data.health:
+            return False
+
+        if self.searchPattern.healthy is False and item.data.health:
+            return False
```

Comment:
```suggestion
        if self.searchPattern.healthy is not None
            return item.data.health == self.searchPattern.healthy
```

This is an optional simplification if I am not mistaken.